### ISSUE TITLE
Hot-swapping a model through /admin/models/load leaves generation producing garbage

### ISSUE THREAD

**Issue by antonellof** [antonellof/ferrox#180]
Hot-swapping the model through `POST /admin/models/load` leaves generation producing garbage, on 0.19.0. The same checkpoint answers correctly when the server is started on it directly, so the file and the engine are fine and the swap path is not.

This is user-visible: the Studio model selector is exactly this endpoint, so switching models in the UI can leave the server answering nonsense until it is restarted.

## Reproduction

Server on `--ngl 99` (Metal), M2 Pro, ferrox 0.19.0, temperature 0, prompt `Capital of Japan? One word.`

| state | active | answer |
|---|---|---|
| fresh start on 3B Q4_K_M | Llama-3.2-3B-Instruct-Q4_K_M | `'Paris.'` for the France prompt, correct |
| swap to 1B Q8_0 | Llama-3.2-1B-Instruct-Q8_0 | `'Tokyo'` correct |
| swap to 3B Q4_K_M | Llama-3.2-3B-Instruct-Q4_K_M | `'Question\nI\n\nThe\n\nThe\n\n is\n\n'` |
| swap to 1B Q8_0 | Llama-3.2-1B-Instruct-Q8_0 | `'Tokyo'` correct |
| swap to 3B Q4_K_M | Llama-3.2-3B-Instruct-Q4_K_M | identical garbage, byte for byte |

Deterministic: the same corrupt string comes back every time, so this is stale state being read, not sampling noise.

The CLI on the same checkpoint is correct:

```
ferrox -m models/Llama-3.2-3B-Instruct-Q4_K_M.gguf -p "..." --ngl 99 --temp 0
```

Q8_0 survives the swap; Q4_K_M does not, at least on this pair.

## The leading hypothesis, to be checked before anything is changed

The repack cache added in #155 and bounded in #158 is keyed on **the host address and length** of the weight matrix. A model swap unmaps one GGUF and mmaps another, and a fresh mmap can be handed the same addresses the old one had. A key that is an address plus a length cannot tell those apart, so a lookup can return a repacked matrix belonging to the previous model.

That would explain every observation: it needs a swap rather than a fresh start, it is deterministic, and it is quant-dependent because the repack tier only covers some kinds.

This is the same defect shape the repo keeps paying for, two structures that must agree with nothing enforcing it, and the fix is presumably an identity a caller cannot get wrong, or dropping the cache on unload.

Note the related precedent: #171 fixed a Metal residency cache that matched on length alone and could hand one request another's activation.

## What settles it

A test that loads model A, swaps to model B, and asserts B's logits match B started fresh. That test should fail today.
